Remove mm_config and mm_license global state from webapp (PR #5)

lieut-data opened the pull request:
#### Summary
This is a subset of the changes in the `MM-8589-remove_mm_global_license` branch, broken up into multiple PRs to facilitate easier review. See any linked PRs for other changes if you're interested. 

These changes, as a whole, move towards removing our dependence on the global `mm_config` and `mm_license` objects. These changes will ultimately facilitate resolving https://mattermost.atlassian.net/browse/MM-8604, allowing us to stop hard refreshing the application whenever someone makes a configuration change.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-8589

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### Related PRs
https://github.com/mattermost/mattermost-webapp/pull/816
https://github.com/mattermost/mattermost-webapp/pull/819
https://github.com/mattermost/mattermost-webapp/pull/820
https://github.com/mattermost/mattermost-webapp/pull/821
https://github.com/mattermost/mattermost-webapp/pull/823

lieut-data commented on components/create_comment/create_comment.jsx:
```diff
@@ -122,7 +122,17 @@ export default class CreateComment extends React.PureComponent {
+         */
+        enableConfirmNotificationsToChannel: PropTypes.bool.isRequired,
+
+        /**
```
This is the type of comment that spawned our conversation on pre-release. For now, these PRs effectively copy the local style of the files being modified, in the hopes that we can go back and holistically rip these out.

lieut-data commented on components/create_team/create_team_controller/index.js:
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
```
Throughout my PRs, I think I copy and pasted from one file to get this copyright notice started. Recognizing that we're going to do a full sweep of these, I'm optimistic this will all get cleaned up later.

ccbrown commented on components/sidebar_right_menu/index.js:
```diff
@@ -15,15 +15,35 @@ import {isMobile} from 'utils/utils.jsx';
+    const helpLink = config.HelpLink;
+    const reportAProblemLink = config.ReportAProblemLink;
+    const restrictTeamInvite = config.RestrictTeamInvite;
+    const siteName = config.SiteName;
```
I may just be out-of-touch when it comes to JavaScript idioms, but it seems like defining these up here more than negates any benefit to using the key-value shorthand below. Is there a reason you don't just define them as key-value pairs below?

hmhealey replied:
I'd personally be a fan of inlining statements like these as long as their purpose is clear. It's really useful to have something complicated like `const isPostVisible = post.visibility == true && post.create_at > lastVisiblePostTime;`, but it feels like we're repeating ourselves with some of these. Only 1/5 though

lieut-data replied:
You're both absolutely right. There were a number of components where the prop computations were non-trivial and all factored out, so I began from that style and needlessly copied it throughout my PRs.

I'm hesitant to go back and change this with half of my PRs merged, but I'll make a note of this and investigate going back and fixing this all in one fell swoop.

lieut-data commented:
I've rebased against master and addressed feedback from parallel reviews (re: using the `get(Config|License)` selectors and eliminating unnecessary `ownProps`).
